x-pack/plugins/inference/scripts/evaluation/scenarios/esql/index.spec.ts

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/// <reference types="@kbn/ambient-ftr-types"/>
+
+import expect from '@kbn/expect';
+import { lastValueFrom } from 'rxjs';
+import { naturalLanguageToEsql } from '../../../../server/tasks/nl_to_esql';
+import { chatClient, evaluationClient, logger } from '../../services';
+
+interface TestCase {
+  title: string;
+  question: string;
+  expected?: string;
+  criteria?: string[];
+  only?: boolean;
+}
+
+interface Section {
+  title: string;
+  tests: TestCase[];
+}
+
+async function evaluateEsqlQuery({
+  question,
+  expected,
+  criteria = [],
+}: {
+  question: string;
+  expected?: string;
+  criteria?: string[];
+}): Promise<void> {
+  logger.debug(`Evaluation: ${question}`);
+
+  const generateEvent = await lastValueFrom(
+    naturalLanguageToEsql({
+      client: {
+        output: chatClient.output,
+        chatComplete: chatClient.chatComplete,
+      },
+      connectorId: chatClient.getConnectorId(),
+      input: question,
+      logger: {
+        debug: (source) => {
+          logger.debug(typeof source === 'function' ? source() : source);
+        },
+      },
+    })
+  );
+
+  logger.debug(`Received response: ${generateEvent.content}`);
+
+  const evaluation = await evaluationClient.evaluate(
+    `# Question
+
+    ${question}
+
+    # Answer
+
+    ${generateEvent.content}
+
+    `,
+    [
+      ...(expected
+        ? [
+            `Returns a ES|QL query that is functionally equivalent to:
+
+            """esql
+            ${expected}
+            """
+
+            It's OK if column names are slightly different, or if the used functions or operators are different,
+            as long as the expected end result is the same.`,
+          ]
+        : []),
+      ...criteria,
+    ]
+  );
+
+  expect(evaluation.passed).to.be(true);
+}
+
+const buildTestDefinitions = (): Section[] => {
+  const testDefinitions: Section[] = [
+    {
+      title: 'ES|QL query generation',
+      tests: [
+        {
+          title: 'Generates a query to show the top 10 domains by doc count',
+          question: `For standard Elastic ECS compliant packetbeat data view (\`packetbeat-*\`),
```

Comment:
We only have 13 tests here, with given the very wide area of things we should be testing, is a very small number. 

I'm not asking to add more in this PR, this is very fine for merging given it's just a port of the existing code from o11y, but I was wondering what the best way to add more tests and to cover more scenarios would be. Like, what was the initial strategy when those tests were added / how were they chosen?

Reply:
these tests have been around for a while and they're probably too focused on the Observability domain, would leave to see us add more. As far as how they were chosen, it's a bit of a mix between "I want the Assistant to be able to do this" and "I see the Assistant makes this mistake".